WalletWasabi.Fluent/ViewModels/Dialogs/InsufficientBalanceDialogViewModel.cs

```diff
@@ -1,29 +1,183 @@
+
+			if (!isInHistory)
+			{
+				RxApp.MainThreadScheduler.Schedule(async () =>
+				{
+					// Edge cases, probably never will happen.
+					if (!EnableSendAnyway && !EnableSubtractFee && !EnableSelectMoreCoin)
+					{
+						await ShowErrorAsync("Transaction Building",
+							"Automatic transaction fee selection is not possible at the moment. Alternatively, you can enter the transaction fee manually in Advanced options.",
+							"Wasabi was unable to create your transaction.");
+						Close(DialogResultKind.Back);
+						return;
+					}
+
+					if (_transactionInfo.IsPayJoin && _wallet.Coins.TotalAmount() == _transactionInfo.Amount)
+					{
+						await ShowErrorAsync("Transaction Building", "There are not enough funds to cover the transaction fee.", "Wasabi was unable to create your transaction.");
+						Close(DialogResultKind.Back);
+						return;
+					}
+
+					IsBusy = true;
+					await CheckSilentCaseAsync(_differenceOfFeePercentage, _wallet, _transactionInfo);
+					SetQuestion();
+					IsBusy = false;
+				});
+			}
 		}
 
-		public string Question { get; }
+		private async Task CheckSilentCaseAsync(decimal percentage, Wallet wallet, TransactionInfo transactionInfo)
+		{
+			if (percentage == 0)
+			{
+				return;
+			}
+
+			if (!TransactionFeeHelper.TryGetMaximumPossibleFee(percentage, wallet, transactionInfo.FeeRate, out var maxFee))
+			{
+				await ShowErrorAsync("Transaction Building",
+					"Automatic transaction fee selection is not possible at the moment. Alternatively, you can enter the transaction fee manually in Advanced options.",
+					"Wasabi was unable to create your transaction.");
+				Close(DialogResultKind.Back);
+				return;
+			}
+
+			_maximumPossibleFee = maxFee;
+			_confirmationTimeWithMaxFee = TransactionFeeHelper.CalculateConfirmationTime(maxFee, wallet);
+
+			if (percentage <= TransactionFeeHelper.FeePercentageThreshold)
+			{
+				OnSendAnyway();
+			}
+		}
+
+		private void SetQuestion()
+		{
+			var sendingTotalBalance = _wallet.Coins.TotalAmount() == _transactionInfo.Amount;
+
+			Question = $"There are not enough funds {(!sendingTotalBalance ? "selected " : "")}to cover the preferred transaction fee. Alternatively, ";
+
+			if (EnableSubtractFee)
+			{
+				Question += "you can subtract the transaction fee from the amount.";
+				return;
+			}
+
+			if (EnableSendAnyway && EnableSelectMoreCoin)
+			{
+				Question += $"you can send the transaction with an adjusted transaction fee, so it should be confirmed within approximately {TextHelpers.TimeSpanToFriendlyString(_confirmationTimeWithMaxFee)} or select more coin.";
+			}
+
+			if (EnableSendAnyway && !EnableSelectMoreCoin)
+			{
+				Question += $"you can send the transaction with an adjusted transaction fee, so it should be confirmed within approximately {TextHelpers.TimeSpanToFriendlyString(_confirmationTimeWithMaxFee)}.";
+			}
+
+			if (!EnableSendAnyway && EnableSelectMoreCoin)
+			{
+				Question += "you can select more coin.";
+			}
```

Comment:
Can be simplified: (Notice that I edited `more coin` to `more coins`)

```suggestion
			if (EnableSendAnyway)
			{
                Question += $"you can send the transaction with an adjusted transaction fee, so it should be confirmed within approximately {TextHelpers.TimeSpanToFriendlyString(_confirmationTimeWithMaxFee)}";

                if (EnableSelectMoreCoin)
				{
                    Question += " or select more coins.";
                }
                else
				{
                    Question += ".";
                }
            }
			else
			{
				if (EnableSelectMoreCoin)
				{
                    Question += "you can select more coins.";
                }
            }
```